scipy/stats/_variation.py

```diff
@@ -110,8 +110,9 @@ def variation(a, axis=0, nan_policy='propagate', ddof=0, *, keepdims=False):
         # The return values, if any, are all nan.
         shp = list(a.shape)
         shp.pop(axis)
-        result = xp.full(shp, fill_value=NaN)
-        return result[()] if result.ndim == 0 else result
+        if shp:
+            return xp.asarray(xp.broadcast_to(NaN, tuple(shp)), copy=True)
```

Comment:
Do we care about the return value to be writeable?

Reply:
Please just pass `full` a `dtype` here and above.